File: uetl/uetl/report/sales_side/sales_side.py
# Copyright (c) 2023, GreyCube Technologies and contributors
# For license information, please see license.txt

import frappe

from uetl.uetl.report.sales_personwise_transaction_ue.sales_personwise_transaction_ue import (
    execute as _execute,
)


def execute(filters=None):
    columns, data = _execute(filters)
    data = data[:-1]

    if filters.get("sez_status") == "Pending":
        data = [d for d in data if d.is_pending_sez]
    elif filters.get("sez_status") == "Completed":
        data = [d for d in data if not d.is_pending_sez]

    for d in data:
        d.billing_month = d.posting_date.strftime("%b")

    return get_columns(columns), data


def get_columns(columns):
    return columns + [
        {
            "label": "Billing Month",
            "fieldname": "billing_month",
            "fieldtype": "Data",
            "width": "140",
        },
        # {
        #     "label": "Parent customer name",
        #     "fieldname": "parent_customer_cf",
        #     "fieldtype": "Data",
        #     "width": "140",
        # },
        # {
        #     "label": "Customer ID",
        #     "fieldname": "customer_id_cf",
        #     "fieldtype": "Data",
        #     "width": "140",
        # },
        {
            "label": "Customer Country",
            "fieldname": "country",
            "fieldtype": "Data",
            "width": "140",
        },
        {
            "label": "Requested ship date (CRD)",
            "fieldname": "tsoi_delivery_date",
            "fieldtype": "Date",
            "width": "130",
        },
        {
            "label": "Payment term (Sales)",
            "fieldname": "payment_terms_template",
            "fieldtype": "Link",
            "options": "Payment Term",
            "width": "190",
        },
        {
            "label": "Batch ID",
            "fieldname": "batch_no",
            "fieldtype": "Link",
            "options": "Batch",
            "width": "140",
        },
        {
            "label": "UOM",
            "fieldname": "uom",
            "fieldtype": "Data",
            "width": "140",
        },
        {
            "label": "Purchaser",
            "fieldname": "purchaser_cf",
            "fieldtype": "Data",
            "width": "140",
        },
        {
            "label": "Is SEZ Pending",
            "fieldname": "custom_sez_file_attachment",
            "fieldtype": "Check",
            "width": "90",
        }
    ]

File: uetl/uetl/report/sales_side/sales_side.js
// Copyright (c) 2022, GreyCube Technologies and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Sales Side"] = {
  filters: [
    {
      fieldname: "sales_person",
      label: __("Sales Person"),
      fieldtype: "Link",
      options: "Sales Person",
    },
    {
      fieldname: "doc_type",
      label: __("Document Type"),
      fieldtype: "Select",
      options: "Sales Invoice",
      default: "Sales Invoice",
    },
    {
      fieldname: "from_date",
      label: __("From Date"),
      fieldtype: "Date",
      default: frappe.defaults.get_global_default("year_start_date"),
    },
    {
      fieldname: "to_date",
      label: __("To Date"),
      fieldtype: "Date",
      default: frappe.datetime.get_today(),
    },
    {
      fieldname: "company",
      label: __("Company"),
      fieldtype: "Link",
      options: "Company",
      default: frappe.defaults.get_user_default("Company"),
      reqd: 1,
    },
    {
      fieldname: "item_group",
      label: __("Item Group"),
      fieldtype: "Link",
      options: "Item Group",
    },
    {
      fieldname: "brand",
      label: __("Brand"),
      fieldtype: "Link",
      options: "Brand",
    },
    {
      fieldname: "customer",
      label: __("Customer"),
      fieldtype: "Link",
      options: "Customer",
    },
    {
      fieldname: "territory",
      label: __("Territory"),
      fieldtype: "Link",
      options: "Territory",
    },
    {
      fieldname: "show_return_entries",
      label: __("Show Return Entries"),
      fieldtype: "Check",
      default: 0,
    },
    {
      fieldname: "cost_center",
      label: __("Cost Center"),
      fieldtype: "Link",
      options: "Cost Center",
    },
    {
      fieldname: "sez_status",
      label: __("SEZ Status"),
      fieldtype: "Select",
      default: 'All',
      width: "120px",
      options: "All\nPending\nCompleted"
    },
  ],

  onload: function () {
    let fiscal_year = erpnext.utils.get_fiscal_year(frappe.datetime.get_today());

    frappe.model.with_doc("Fiscal Year", fiscal_year, function (r) {
      var fy = frappe.model.get_doc("Fiscal Year", fiscal_year);
      frappe.query_report.set_filter_value({
        from_date: fy.year_start_date,
      });
    });
  },
};
